src/routes/manage_job/post.rs

```diff
@@ -0,0 +1,264 @@
+//! This file implements the `POST /bulk` endpoint.
+use crate::{
+	errors::ReacherError,
+	routes::check_email::{header::check_header, post::RetryOption},
+};
+use async_recursion::async_recursion;
+use check_if_email_exists::{
+	check_email as ciee_check_email, CheckEmailInput, CheckEmailInputProxy, CheckEmailOutput,
+	Reachable,
+};
+use sqlx::{Pool, Postgres};
+use warp::Filter;
+
+use std::{collections::HashMap, error::Error};
+
+use crate::routes::check_email::known_errors;
+use serde::{Deserialize, Serialize};
+use sqlxmq::{job, CurrentJob};
+
+const EMAIL_TASK_BATCH_SIZE: usize = 1;
+
+/// Errors that can happen during an email verification.
+#[derive(Debug)]
+enum CheckEmailError {
+	/// We get an `is_reachable` Unknown. We consider this internally as an
+	/// error case, so that we can do retry mechanisms (see select_ok & retry).
+	Unknown((CheckEmailOutput, RetryOption)),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct TaskInput {
+	job_id: i32,
+	input: CheckEmailInput,
+}
+
+/// Endpoint request body.
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub struct CreateBulkRequestBody {
+	input_type: String,
+	input: Vec<String>,
+	proxy: Option<HashMap<String, String>>,
```

Comment:
```suggestion
	proxy: Option<CheckEmailInputProxy>,
```

from check_if_email_exists::CheckEmailInputProxy